driver/csiplugin/gpfs_util.go

```diff
@@ -381,3 +430,76 @@ func numberInSlice(a int, list []int) bool {
 	}
 	return false
 }
+
+func getVolIDMembers(vID string) (scaleVolId, error) {
+	splitVid := strings.Split(vID, ";")
+	var vIdMem scaleVolId
+
+	if len(splitVid) == 3 {
+		/* This is LW volume */
+		/* <cluster_id>;<filesystem_uuid>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		SlnkPart := splitVid[2]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = false
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 4 {
+		/* This is fileset Based volume */
+		/* <cluster_id>;<filesystem_uuid>;fileset=<fileset_id>;path=<symlink_path> */
+		vIdMem.ClusterId = splitVid[0]
+		vIdMem.FsUUID = splitVid[1]
+		fileSetPart := splitVid[2]
+		fileSetSplit := strings.Split(fileSetPart, "=")
+		if len(fileSetSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+
+		if fileSetSplit[0] == "filesetName" {
+			vIdMem.FsetName = fileSetSplit[1]
+		} else {
+			vIdMem.FsetId = fileSetSplit[1]
+		}
+
+		SlnkPart := splitVid[3]
+		slnkSplit := strings.Split(SlnkPart, "=")
+		if len(slnkSplit) < 2 {
+			return scaleVolId{}, status.Error(codes.Internal, fmt.Sprintf("Invalid Volume Id : [%v]", vID))
+		}
+		vIdMem.SymLnkPath = slnkSplit[1]
+		vIdMem.IsFilesetBased = true
+		return vIdMem, nil
+	}
+
+	if len(splitVid) == 8 {
+		/* Volume ID created from 2.5.0 onwards  */
+		/* VolID: <storageclass_type>;<type_of_volume>;<cluster_id>;<filesystem_uuid>;<consistency_group>;<application>;<fileset_name>;<symlink_path> */
```

Comment:
What is the use of having consistencygroup vs application parameter in volume handle ? I do assume that application might have multiple consistencygroups but at storage level we would be managing consistency group as a single entity only on which user should be able to perform group level operation such as create snapshot etc.

Reply:
I think this would be useful in use cases like creating replicas or taking snapshots of particular (or all) CGs of an application, but @deeghuge  can comment more from design perspective.

Reply:
I had discussion with Abhishek yesterday. I am inclined towards removing the application from volume handle since we don't know the exact requirement for future use. @amdabhad can you take care changing volume handle to 
`VolID: <storageclass_type>;<type_of_volume>;<cluster_id>;<filesystem_uuid>;<consistency_group>;<fileset_name>;<symlink_path>`

@madhuthorat FYI

Reply:
Done